plugins/janus_sip.c

```diff
@@ -4123,14 +4133,22 @@ static void *janus_sip_handler(void *data) {
 						/* Take note of the original media direction */
 						session->media.pre_hold_video_dir = m->direction;
 						/* Update the media direction */
-						switch(m->direction) {
-							case JANUS_SDP_DEFAULT:
-							case JANUS_SDP_SENDRECV:
-								m->direction = JANUS_SDP_SENDONLY;
-								break;
-							default:
+						if(media_direction_text != NULL && strlen(media_direction_text) > 0){
+							if(!strcasecmp(media_direction_text, "inactive")){
 								m->direction = JANUS_SDP_INACTIVE;
-								break;
+							} else {
+								m->direction = JANUS_SDP_SENDONLY;
+							}
+						} else {
+							switch(m->direction) {
+								case JANUS_SDP_DEFAULT:
+								case JANUS_SDP_SENDRECV:
+									m->direction = JANUS_SDP_SENDONLY;
+									break;
+								default:
+									m->direction = JANUS_SDP_INACTIVE;
+									break;
+							}
```

Comment:
Both these changes are overkill, IMHO. They could be left exactly as they are, where only `JANUS_SDP_SENDONLY` is parametrized as a variable. There's no need to do a `strcasecmp` on the argument: there's a method in the SDP utils, called `janus_sdp_parse_mtype`, that can parse it to a `janus_sdp_mtype`, meaning it can be used directly there with minimal changes.